client/AccountsManagerClient.ts

```diff
@@ -42,12 +43,20 @@ export default class AccountsManagerClient {
         this._redirectClient = new RedirectRpcClient('', RequestBehavior.getAllowedOrigin(this._endpoint));
     }
 
-    public init() {
+    public checkRedirectResponse() {
         return this._redirectClient.init();
     }
 
-    public on(command: RequestType, resolve: (...args: any[]) => any, reject: (...args: any[]) => any) {
-        this._redirectClient.onResponse(command, resolve, reject);
+    public on(
+        command: RequestType,
+        resolve: (result: RpcResult, state: any) => any,
+        reject?: (error: Error, state: any) => any
+    ) {
+        this._redirectClient.onResponse(command,
+            // State is always an object containing at least the __command property
+            (result: RpcResult, rpcId, state) => resolve(result, JSON.parse(state!)),
+            (error: Error, rpcId, state) => reject && reject(error, JSON.parse(state!)),
```

Comment:
I think we agreed that state will always be a string now. I'm OK with changing though. Final?

Reply:
I found out that it actually cannot be a string, specifically because of this line:
https://github.com/nimiq/accounts/blob/master/client/RequestBehavior.ts#L58

I talked with Pascal and he agreed that it won't work, thus I decided to just go with a plain object instead and added the de-serialization wrapper.